### PR TITLE
BUG: Fix Fortran kind detection for aarch64 & s390x.

### PR DESCRIPTION
**Pull request by QuLogic**
See https://github.com/numpy/numpy/issues/3424#issuecomment-412327556 and #8659.

### CODE REVIEW

**Comment by QuLogic** [comment]
Also, cc @tylerjereddy because I'm unsure why this does not fail on Shippable?

File: numpy/f2py/crackfortran.py [hunk 1]
```diff
@@ -2403,7 +2403,7 @@ def _selected_real_kind_func(p, r=0, radix=0):
     if p < 16:
         return 8
     machine = platform.machine().lower()
-    if machine.startswith('power') or machine.startswith('ppc64'):
+    if machine.startswith(('aarch64', 'power', 'ppc64', 's390x')):
```
Comment (QuLogic) [inline numpy/f2py/crackfortran.py]: Also, not sure if this should be a strict check for the new ones instead of `startswith`.

Comment (charris) [reply numpy/f2py/crackfortran.py]: Probably more compiler dependent, I'd guess. Let's put this in and fix it up if problems ensue.

**Comment by charris** [comment]
Thanks @QuLogic .

**Pull request merged**